### ISSUE TITLE
No display output on Quest headsets with SteamVR runtime

### ISSUE THREAD

**Issue by CherrySodaPop** [GodotVR/godot_openxr#121]
There's no output on the Quest 2 headset when using a wired link but the desktop window does clearly show the player moving along side their hands moving.

I've also tested this with a WMR headset with it displaying correctly.

![image](https://user-images.githubusercontent.com/42105283/141660692-3286ac83-d5ff-4991-a130-98e7dbf66e4a.png)
![image](https://user-images.githubusercontent.com/42105283/141660695-ab59c2a1-4cf7-4fa7-9dbf-0459cff29726.png)

**Comment by BastiaanOlij** [comment]
@m4gr3d any ideas what could be causing this?

Just tried with the Quest 1. This works for me when using Oculus OpenXR runtime, but going through SteamVR I have the same issue. This used to work for me (though I admit I mostly tested SteamVR with Rift S).

@CherrySodaPop can you see if you can get the same behavior? If you go to project settings and make sure to enable the plugin you get a dropdown in Godot to switch between the different OpenXR runtimes. Select the Oculus OpenXR Runtime here.
Do note that Oculus does not start automatically, you have to run the Oculus app on your PC first so that link is established.

**Comment by CherrySodaPop** [comment]
Got the same behavior, works perfectly with Oculus but nothing on SteamVR.
Should've tested the diffrent runtimes 😅 glad knowing I can work around this though.

**Comment by isosphere** [comment]
@CherrySodaPop I have had similar trouble using Air Link w/ the Oculus Quest 2 for games in general, and have developed something of a process that seems to mostly work. Maybe it'll change your results?:

1. Start link
2. Launch SteamVR
3. Minimize SteamVR via Oculus button
4. Launch target application
5. If necessary, switch to new task via Oculus Link bar (minimize, scroll to right-most area of bar, find plus, find target application)

**Comment by CherrySodaPop** [comment]
@isosphere Your method unfortuantly doesn't seem to work for me.
It continues to show the "Next up" in SteamVR.

**Comment by m4gr3d** [comment]
Should the bug title be updated? Based on the discussion it seems to be a steamvr issue rather than being specific to the Quest 2 (with link).

**Comment by gattila** [comment]
I have a similar issue with Quest2/SteamVR

**Comment by CarterTMDev** [comment]
I am having the same issue with Quest 1 using SteamVR via the Virtual Desktop app (downloaded from the Oculus Quest Store). This setup works with other SteamVR games, as well as with the OpenVR Godot plugin, just not with the OpenXR Godot plugin. Here's my console output:

```
--- Debugging process started ---
Godot Engine v3.4.stable.mono.official.206ba70f4 - https://godotengine.org
OpenGL ES 3.0 Renderer: Radeon (TM) RX 480 Graphics
OpenGL ES Batching: ON
 
Running on OpenXR runtime: SteamVR/OpenXR 0.1.0
OpenXR Runtime only supports OpenGL version 4.3 - 4.6!
OpenXR Using OpenGL version: 3.3.14761 Core Profile Forward-Compatible Context 21.30.23.01 30.0.13023.1012
OpenXR Using OpenGL renderer: Radeon (TM) RX 480 Graphics
OpenXR Interface initialized
No refresh rate given by XR runtime
OpenXR EVENT: session state changed to XR_SESSION_STATE_IDLE
On state idle
OpenXR EVENT: session state changed to XR_SESSION_STATE_READY
On state ready
OpenXR Swapchain Formats
OpenXR Couldn't find prefered swapchain format, using 32859
Attached action set godot
OpenXR found internal action aim_pose
OpenXR found internal action grip_pose
OpenXR found internal action front_trigger
OpenXR found internal action side_trigger
OpenXR found internal action primary
OpenXR found internal action secondary
OpenXR found internal action ax_button
OpenXR found internal action by_button
OpenXR found internal action ax_touch
OpenXR found internal action by_touch
OpenXR found internal action menu_button
OpenXR found internal action select_button
OpenXR found internal action front_button
OpenXR found internal action front_touch
OpenXR found internal action side_button
OpenXR found internal action primary_button
OpenXR found internal action secondary_button
OpenXR found internal action primary_touch
OpenXR found internal action secondary_touch
OpenXR found internal action haptic
Hand tracking is supported

OpenXR session begun
OpenXR EVENT: session state changed to XR_SESSION_STATE_SYNCHRONIZED
On state synchronized
OpenXR EVENT: session state changed to XR_SESSION_STATE_VISIBLE
On state visible
OpenXR visible state
OpenXR EVENT: session state changed to XR_SESSION_STATE_FOCUSED
On state focused
OpenXR focused state
OpenXR mapped /user/hand/left to 1
OpenXR regained tracking for hand 0
OpenXR mapped /user/hand/right to 2
OpenXR regained tracking for hand 1
Activated LeftHandController
Activated RightHandController
OpenXR EVENT: interaction profile changed!
OpenXR EVENT: interaction profile changed!
OpenXR EVENT: interaction profile changed!
OpenXR EVENT: interaction profile changed!
OpenXR EVENT: interaction profile changed!
OpenXR Event: Interaction profile changed for /user/hand/left: /interaction_profiles/oculus/touch_controller
OpenXR Event: Interaction profile changed for /user/hand/right: /interaction_profiles/oculus/touch_controller
OpenXR EVENT: interaction profile changed[...]
 [output overflow, print less text!]
OpenXR EVENT: session state changed to XR_SESSION_STATE_VISIBLE
On state visible
OpenXR visible state
OpenXR EVENT: session state changed to XR_SESSION_STATE_SYNCHRONIZED
On state synchronized
OpenXR EVENT: session state changed to XR_SESSION_STATE_STOPPING
On state stopping
OpenXR session ending
OpenXR EVENT: session state changed to XR_SESSION_STATE_IDLE
On state idle
OpenXR EVENT: session state changed to XR_SESSION_STATE_EXITING
On state exiting
Deactivated LeftHandController
Deactivated RightHandController
--- Debugging process stopped ---
```
For some reason it couldn't get the Quest's refresh rate (72 Hz). Also, there's that warning `OpenXR Runtime only supports OpenGL version 4.3 - 4.6!` I'm running the Godot project as GLES3 (AKA OpenGL 3.3). Could that have something to do with it?

**Comment by hohfchns** [comment]
Also having this issue on Oculus Quest 1 with SteamVR. Would love an update on this

**Comment by CherrySodaPop** [comment]
Update! Release 1.2.0 seems to have fixed the issue.
Oculus and SteamVR runtimes both work now.
![image](https://user-images.githubusercontent.com/42105283/154856189-a1c8f48a-743d-4a73-b709-3a0784705a1b.png)

**Comment by m4gr3d** [comment]
Seems the issue was fixed in release 1.2.0.
